### PR TITLE
added and modified a function in sympy/tensor/index_methods.py to preserve order of indices

### PR DESCRIPTION
**Pull request by avishrivastava11**
Added `_unique_and_repeated` method and modified `_remove_repeated` method in `sympy/tensor/index_methods.py`

Took help and reviews from #14790 

Added a `_unique_and_repeated` method which returns the unique and repeated indices. This is used in the function `_remove_repeated`  which calls it to  remove the repeated indices and make them occur once only as also shown in the example.

Note : I have taken the review given by @smichr in PR #14790 and made this PR (since I found this particularly helpful and no-one else was addressing it)

Pls review this PR and let me know the improvements.

### CODE REVIEW

File: sympy/tensor/index_methods.py [hunk 1]
```diff
@@ -23,27 +23,43 @@
 class IndexConformanceException(Exception):
     pass
 
+def _unique_and_repeated(inds):
+    """returns the unique and repeated indices. This is used in the function
```
Comment (smichr) [inline sympy/tensor/index_methods.py]: At the top of this file you need to import OrderedDict.

In the docstring, it is not important to say who uses this helper. That will be self-evident from the codebase now and in the future. You probably should indicate that the order of indices is preserved, however, and use a simple example like `[2, 3, 1, 3, 0, 4, 0] -> [2, 1, 4], [3, 0]`.

Comment (avishrivastava11) [reply sympy/tensor/index_methods.py]: @smichr  Thanks for the review, I have made a few changes. Pls take a look and suggest improvement(s)

**Comment by avishrivastava11** [comment]
@asmeurer @smichr  can you review this PR ? and also, can you re-run the travis-build ? (I think there must've been some problem with my build, not the code, and that's why one test is having an issue)

File: sympy/tensor/index_methods.py (hunk 1 above)
Comment (avishrivastava11) [reply sympy/tensor/index_methods.py]: @smichr  also, can you re-run the Travis build?
I think there was an issue with one test-run, not the code and that's why one of the test is giving an error.

**Comment by avishrivastava11** [comment]
@jksuom @smichr @certik Can you review this PR ?

**Comment by avishrivastava11** [comment]
Thanks @smichr for merging :) 
Really appreciate it :))

**Pull request merged**